cmake/FindGcov.cmake

```diff
@@ -141,9 +141,9 @@ function (add_gcov_target TNAME)
 
 		# call gcov
 		add_custom_command(OUTPUT ${TDIR}/${FILE}.gcov
-			COMMAND ${GCOV_ENV} ${GCOV_BIN} ${TDIR}/${FILE}.gcno > /dev/null
+			COMMAND ${GCOV_ENV} ${GCOV_BIN} -p ${TDIR}/${FILE}.gcno > /dev/null
 			DEPENDS ${TNAME} ${TDIR}/${FILE}.gcno
-			WORKING_DIRECTORY ${FILE_PATH}
```

Comment:
I think, this change can lead to name clashes. Is this change specifically necessary for ninja support?

Reply:
There is no file name conflict, although the WORKING_DIRECTORY is always CMAKE_CURRENT_BINARY_DIR, the -p flag ensures that the generated .gcov file name contains the source file path, such as The gcov file generated from ../src/main.cpp is called ^#src#main.cpp.gcov.